ui/apps/dashboard/src/app/(organization-active)/(dashboard)/settings/integrations/vercel/enrichVercelProjects.ts

```diff
@@ -1,18 +1,8 @@
-import { graphql } from '@/gql';
 import graphqlAPI from '@/queries/graphqlAPI';
 import { getProductionEnvironment } from '@/queries/server-only/getEnvironment';
 import { type VercelProject, type VercelProjectViaAPI } from './VercelIntegration';
-
-const GetSavedVercelProjectsDocument = graphql(`
-  query GetSavedVercelProjects($environmentID: ID!) {
-    environment: workspace(id: $environmentID) {
-      savedVercelProjects: vercelApps {
-        projectID
-        path
-      }
-    }
-  }
-`);
+import mergeVercelProjectData from './mergeVercelProjectData';
+import { GetSavedVercelProjectsDocument } from './queries';
```

Comment:
WIP. Extracted re-usable logic here as to not have to refactor everything away from RSC in one shot on initial commit.